aten/src/ATen/NestedTensorImpl.h

```diff
@@ -215,6 +224,10 @@ inline NestedTensorImpl* get_nested_tensor_impl(const at::Tensor& tensor) {
 }
 
 inline bool nested_tensor_impl_is_contiguous(const NestedTensorImpl* nt) {
+  if (nt->is_meta()) {
+    // Simplify a lot of checks by claiming contiguous for meta tensors.
```

Comment:
Not sure how this propgate through entirely but there is no reason why meta nts need be contiguous right? I.e the meta tensor output from nt.transpose also should the contiguous

Reply:
Yeah this is not right. However, I am sympathetic to the intent of the change here, which is that for the E2E model we're targeting, none of the fancy discontiguous nested tensor stuff is being used. So actually SymInt'ifying this code would constitute a bunch of unnecessary work that doesn't actually contribute to anything. I'm not exactly sure what the best way to thread the needle here is though.

Reply:
Like, has discontiguous NT actually been used for anything real? Can we delete it entirely?

Reply:
Yes, unfortunately the math path for SDPA does do transposes that result in discontiguous NTs used further down..

Reply:
The issue is that contiguous checks are littered throughout NT code, in places like `clone()`, `empty_like()`, and anywhere `wrap_buffer()` is used (including the main T -> NT entrypoint). There's no way to know if a meta NT is contiguous because the strides are symbolic.

The hack here gets around this, but I don't think it's the right way to go about it.

Reply:
Okay, I got around this by removing the general hack here and just checking against meta for a few cases that call this:
* `get_buffer()` - don't check contiguity here for meta; just return the buffer
* `empty_like_nested()` -> `verify_empty_parameters()` - also don't check contiguity here for meta

This is enough to get things passing. cc @drisspg @cpuhrsch

Reply:
Punting this to the future, but as discussed, sometimes we can know if a meta NT is contiguous: if `nested_strides` / `offsets` are possibly derived from `nested_sizes`, we can check if they are "defined" and assume that if they're not, the NT is contiguous. We could also propagate contiguity across op calls if the contiguity of the output is deterministically inferrable from the contiguity of the input.

Anyway, I sidestepped this for now and I don't think it matters for the two call-sites I mentioned above.